src/main/java/de/tum/in/www1/artemis/web/rest/StudentExamResource.java

```diff
@@ -574,9 +596,17 @@ private void setResultIfNecessary(StudentExam studentExam, StudentParticipation
                 // lastResult -> lastSubmission
                 participation.setResults(Set.of(latestResult));
             }
+            else {
+                // To prevent LazyInitializationException.
+                participation.setResults(Set.of());
```

Comment:
Could you set the results to the empty set once before the outer `if (isStudentAllowedToSeeResult …)`? Then the cases where it has to be overwritten can remain as-is, and you can remove the `else`-cases.
That would also make it less error-prone in case of future changes.

Reply:
Good catch! I added the `participation.setResults(Set.of());` lines on different occasions when they caused errors so I didn't noticed your suggestion is equivalent. I changed it.